### ISSUE TITLE
PRIVATE link dependencies are not converted to _unity format

### ISSUE THREAD

**Issue by jcelerier** [sakra/cotire#85]
Hi,

when a target is marked PRIVATE when linking : 

```
target_link_libraries(b_lib PRIVATE a_lib)
```

then calling 

```
make b_lib_unity
```

builds `a_lib` (non-unity) (with both make and ninja).

I attached an example.

[bug.zip](https://github.com/sakra/cotire/files/93357/bug.zip)

**Comment by sakra** [comment]
You have to do the creation of unity targets for underlying libraries yourself. See the section "automatically setting up linked libraries in the unity target" in the cotire manual.

**Comment by jcelerier** [comment]
`a_lib` in my case is a library in the same project.

And with COTIRE_DEBUG I get for `PRIVATE` linking : 

```
-- unity target b_lib_unity link strategy: COPY_UNITY
-- unity target b_lib_unity libraries: a_lib_unity;$<LINK_ONLY:a_lib>
```

and for PUBLIC linking:
 
```
-- unity target b_lib_unity link strategy: COPY_UNITY
-- unity target b_lib_unity libraries: a_lib_unity
```

So the `$<LINK_ONLY:a_lib>` seems to be the culprit

**Comment by sakra** [comment]
Should be fixed in 1.7.8.
